hashicorp/dev-portal
# fix: package-manager code block copy

## 🔗 Relevant links

- [Preview link][preview] 🔎
- [Asana task][task] 🎟️

## 🗒️ What

Fixes an issue where install commands on the download views were very difficult, if not impossible, to copy. Separate issues affected the ability to copy by selecting text and the ability to copy with the code-block "Copy" button.

## 🤷 Why

To ensure practitioners can easily copy and use the install commands on all downloads views.

## 🛠️ How

- Bumps to new version of code-block that supports multiple single-line commands in a single `shell-snippet`
	- See https://github.com/hashicorp/react-components/pull/736 for details
- Refactors install-view `packageManagers` code to ensure the code is highlighted, by moving some work previously done on the client to the server.
	- This is necessary to fix the select-copy-paste functionality (in the way supported by the `code-block` release above). We use CSS to target the syntax-highlighted `shell-symbol` elements (namely `$`); so highlighting must be applied for the fix to work.


## 🧪 Testing

- [ ] Visit a downloads view, such as [/terraform/downloads](https://dev-portal-git-zsfix-install-view-code-block-copy-hashicorp.vercel.app/terraform/downloads)
	- For any of the "install command" code blocks for any operating system...
	- Select the code, copy it, and paste into a plain text editor. The install commands should have pasted with the correct newlines, and without leading `$` characters.
		- Note that there may be leading spaces with this method; tweaking this would further refinement outside the scope of this PR.
	- Activate the "Copy" button, and paste into a plain text editor. The install commands should have pasted with the correct newlines, and without leading `$` characters or leading spaces.

## 💭 Anything else?

Not at the moment!

[task]: https://app.asana.com/0/1202097197789424/1203067508838442/f
[preview]: https://dev-portal-git-zsfix-install-view-code-block-copy-hashicorp.vercel.app/terraform/downloads
